ros2param/ros2param/verb/dump.py

```diff
@@ -93,17 +80,24 @@ def main(self, *, args):  # noqa: D102
                     'Exception while calling list_parameters service of node '
                     f"'{node_name.full_name}': {e}", file=sys.stderr)
                 return
+            parameter_names = sorted(response.result().result.names)
 
-            response = response.result().result.names
-            response = sorted(response)
-            parameter_values = self.get_parameter_values(node, absolute_node_name, response)
-            if parameter_values is None:
+            # retrieve parameter values
+            try:
+                response = call_get_parameters(
+                    node=node, node_name=absolute_node_name, parameter_names=parameter_names)
+            except RuntimeError as e:
                 print(
                     'Exception while calling get_parameters service of node '
                     f"'{node_name.full_name}': {e}", file=sys.stderr)
-                return
-
-            for param_name, pval in zip(response, parameter_values):
+            if response.values is None:
```

Comment:
I think this line of code is a bug.  If `call_get_parameters` throws an exception, then we'll fall through to here, but `response` isn't assigned to anything.  So I think we'll end up with a `NameError`.  Maybe what we should do is set a variable called `response_values = None` before `call_get_parameters`, and then only if call_get_parameters succeeds assign result.values to `result_values`.

Reply:
right, this is bug. addressed in https://github.com/ros2/ros2cli/pull/933/commits/bb452ccbebc630528c27a721ebd388aff019cbca